src/modules/rf/rf.cpp

```diff
@@ -1094,3 +1124,220 @@ bool txSubFile(FS *fs, String filepath) {
   return true;
 }
 
+// Static array of sub-GHz frequencies in MHz
+static const float subghz_frequency_list[] = {
+  /* 300 - 348 MHz Frequency Range */
+  300.000f, 302.757f, 303.875f, 303.900f, 304.250f,
+  307.000f, 307.500f, 307.800f, 309.000f, 310.000f,
+  312.000f, 312.100f, 312.200f, 313.000f, 313.850f,
+  314.000f, 314.350f, 314.980f, 315.000f, 318.000f,
+  330.000f, 345.000f, 348.000f, 350.000f,
+
+  /* 387 - 464 MHz Frequency Range */
+  387.000f, 390.000f, 418.000f, 430.000f, 430.500f,
+  431.000f, 431.500f, 433.075f, 433.220f, 433.420f,
+  433.657f, 433.889f, 433.920f, 434.075f, 434.177f,
+  434.190f, 434.390f, 434.420f, 434.620f, 434.775f,
+  438.900f, 440.175f, 464.000f, 467.750f,
+
+  /* 779 - 928 MHz Frequency Range */
+  779.000f, 868.350f, 868.400f, 868.800f, 868.950f,
+  906.400f, 915.000f, 925.000f, 928.000f
+};
+
+void rf_scan_copy() {
+    if (!initRfModule("rx")) {
+        return;
+    }
+
+    RfCodes received;
+    RCSwitch rcswitch = RCSwitch();
+
+    if (RfModule == 1) {
+        #ifdef USE_CC1101_VIA_SPI
+            #ifdef CC1101_GDO2_PIN
+                rcswitch.enableReceive(CC1101_GDO2_PIN);
+            #else
+                rcswitch.enableReceive(CC1101_GDO0_PIN);
+                Serial.println("CC1101 enableReceive()");
+            #endif
+        #else
+            return;
+        #endif
+    } else {
+        rcswitch.enableReceive(RfRx);
+    }
+
+    if (RfScanRange < 0 || RfScanRange > 3) {
+        RfScanRange = 3;
+    }
+
+    const char* sz_range[] = {"300-348 MHz", "387-464 MHz", "779-928 MHz", "All ranges" };
+
+    int range_limits[][2] = {
+        { 0, 23 },  // 300-348 MHz
+        { 24, 47 }, // 387-464 MHz
+        { 48, 56 }, // 779-928 MHz
+        { 0, sizeof(subghz_frequency_list) / sizeof(subghz_frequency_list[0]) - 1} // All ranges
+    };
+
+RestartScan:
+    drawMainBorder();
+    tft.setCursor(10, 28);
+    tft.setTextSize(FP);
+    tft.println("Waiting for signal.");
+    tft.setCursor(10, tft.getCursorY());
+    if (RfFxdFreq) {
+        tft.println("Freq: " + String(RfFreq) + " MHz");
+    }
+    else {
+        tft.println("Range: " + String(sz_range[RfScanRange]));
+    }
+    tft.setCursor(10, tft.getCursorY()+LH*2);
+    tft.println("Press [NEXT] for range.");
+
+    char hexString[64];
+    int idx = range_limits[RfScanRange][0];
+
+    float found_freq = 0.f;
+
+    for (;;) {
+        if (idx < range_limits[RfScanRange][0] || idx > range_limits[RfScanRange][1]) {
+            idx = range_limits[RfScanRange][0];
+        }
+
+        if (checkEscPress()) {
+            break;
+        }
+
+        float frequency = RfFxdFreq ? RfFreq : subghz_frequency_list[idx];
+        
+        ELECHOUSE_cc1101.setMHZ(frequency);
+        delay(50);
```

Comment:
This delay is preventing the frequency detection to work properly.. it will take 24x50ms to run all frequencies in the 300-348Mhz list for example (1200 milliseconds).. and we can't detect anything unless you keep pressing the btn for this long, and my gate controller is beig detecte in more than one frequency (may be untestandable tho, not a real issue, maybe)

I can't test sending the singals from Cardputer in other ranges, becaus it is not being detected..

Can it fix the frequency (for the scan session) after detect the frequency? it would be interesting, becaus once it gets a signal, it can get half of the code or somthing.. so you can resend watching the frequency you've got,... did I made myself clear?

call me on discord to talk about it (Pirata)